docs/docs/user-guide.md

```diff
@@ -4,164 +4,219 @@ sidebar_position: 2
+are *not* hierarchical; applying `@NullMarked` to package `com.foo` does not
+make package `com.foo.bar` `@NullMarked`.) If applied to a class, interface, or
+method, then its scope is all the code in that class, interface, or method.
 
 ```java
 @NullMarked
-public class Strings {
-  public static String nullToEmpty(@Nullable String x) {
-    return (x == null) ? "" : x;
+class Strings {
+  static @Nullable String emptyToNull(String x) {
+    return x.isEmpty() ? null : x;
   }
 
-  public static int spaceIndex(String x) {
-    return x.indexOf(' ');
+  static String nullToEmpty(@Nullable String x) {
+    return x == null ? "" : x;
   }
 }
 ```
 
-In this example, both methods are in the scope of `@NullMarked`, so plain
-`String` means "a reference to a string object, not null". `@Nullable String`
-continues to mean "a reference to a string object, or null". Tools should warn
-you if you try to pass a "reference to a string object, or null" to
-`spaceIndex`, since its argument can't be null, and indeed it will throw
-NullPointerException if given a null argument.
-
-As mentioned above, there are some exceptions to this interpretation for local
-variables (as we'll see next) and [type variables](#defining-generics).
-
-## Local variables {#local-variables}
-
-Tools that understand JSpecify annotations typically *don't* require `@Nullable`
-to be applied to local variables. They may in fact not even allow that. The
-reason is that it should be possible to *infer* whether a variable can be null
-based on the values that are assigned to the variable. For example:
+Here's the example from above, where the class containing the methods is
+annotated with `@NullMarked`. The nullness of the types is the same as before:
+`emptyToNull` does not accept `null` arguments, but it might return `null`;
+`nullToEmpty` does accept `null` arguments, but it won't return `null`. But we
+were able to do that with fewer annotations. In general, using `@NullMarked`
+will give you correct nullness semantics with fewer annotations. In
+`@NullMarked` code, you'll get used to thinking about plain, unannotated types
+like `String` as meaning a real reference to a `String` object and never `null`.
+
+As mentioned above, there are some exceptions to this interpretation for
+[local variables](#local-variables) and [type variables](#declaring-generics).
+
+### `@NullUnmarked`
+
+If you're applying JSpecify annotations to your code, you might not be able to
+annnotate it all at once. If you can apply `@NullMarked` to some of your code
+now, and do the rest later, that's better than waiting until you have time to
+annotate everything. But that means you may have to null-mark a module, package
+or class *except for some classes or methods*. To do that, apply
+[`@NullUnmarked`] to a package, class, or method that's already inside a
+`@NullMarked` span. `@NullUnmarked` simply undoes the effects of the surrounding
```

Comment:
I think we've been using "context" and "scope" over "span." (I personally like "scope" because context is One of Those Words that gets used for everything, like "manager," but "context" is fine with me if you prefer it.)